server/manager/src/test/java/org/apache/accumulo/manager/upgrade/Upgrader11to12Test.java

```diff
@@ -62,198 +67,334 @@ public class Upgrader11to12Test {
+    Text row1 = new Text("123");
+
+    String fileName1 = "hdfs://localhost:8020/accumulo/tables/12/default_tablet/A000000v.rf";
+    Key key1 =
+        Key.builder(false).row(row1).family(DataFileColumnFamily.NAME).qualifier(fileName1).build();
+    Value value1 = new Value("123,456");
+    scanData.put(key1, value1);
+
+    String fileName2 = "hdfs://localhost:8020/accumulo/tables/12/default_tablet/B000000v.rf";
+    Key key2 =
+        Key.builder(false).row(row1).family(DataFileColumnFamily.NAME).qualifier(fileName2).build();
+    Value value2 = new Value("321,654");
+    scanData.put(key2, value2);
+
+    Key chop1 = Key.builder(false).row(row1).family(ChoppedColumnFamily.NAME)
+        .qualifier(ChoppedColumnFamily.NAME).build();
+    scanData.put(chop1, null);
+
+    Key extern1 = Key.builder(false).row(row1).family(ExternalCompactionColumnFamily.NAME)
+        .qualifier(ExternalCompactionColumnFamily.NAME).build();
+    scanData.put(extern1, null);
 
-    Key k = Key.builder().row(new Text("12;")).family(DataFileColumnFamily.NAME)
-        .qualifier(stf.getMetadataText()).build();
-    Value v = new Value("1234,5678");
+    Text row2 = new Text("234");
 
-    upgrader.upgradeDataFileCF(k, v, bw, "aTable");
+    String fileName3 = "hdfs://localhost:8020/accumulo/tables/13/default_tablet/C000000v.rf";
+    Key key3 =
+        Key.builder(false).row(row2).family(DataFileColumnFamily.NAME).qualifier(fileName3).build();
+    Value value3 = new Value("1,2");
+    scanData.put(key3, value3);
 
-    // with file entry in correct formation, no mutations are expected.
-    verify(bw);
-  }
+    Scanner scanner = mock(Scanner.class);
+    scanner.fetchColumnFamily(DataFileColumnFamily.NAME);
+    expectLastCall();
+    scanner.fetchColumnFamily(ChoppedColumnFamily.NAME);
+    expectLastCall();
+    scanner.fetchColumnFamily(ExternalCompactionColumnFamily.NAME);
+    expectLastCall();
+
+    expect(scanner.iterator()).andReturn(scanData.entrySet().iterator()).once();
+    replay(batchWriter, scanner);
 
-  @Test
-  void upgradeDataFileCFInvalidMutationTest() throws Exception {
     Upgrader11to12 upgrader = new Upgrader11to12();
+    upgrader.processReferences(batchWriter, scanner, "accumulo.metadata");
 
-    BatchWriter bw = createMock(BatchWriter.class);
-    Capture<Mutation> capturedAdd = newCapture();
-    bw.addMutation(capture(capturedAdd));
-    expectLastCall().andThrow(new MutationsRejectedException(null, List.of(), Map.of(), List.of(),
-        0, new NullPointerException()));
+    LOG.info("c:{}", capturedUpdate1.getValue().prettyPrint());
+    var u1 = capturedUpdate1.getValue();
+    // 2 file add, 2 file delete. 1 chop delete, 1 ext comp delete
+    assertEquals(6, u1.getUpdates().size());
 
-    replay(bw);
+    LOG.info("c:{}", capturedUpdate2.getValue().prettyPrint());
+    var u2 = capturedUpdate2.getValue();
+    // 1 add, 1 delete
+    assertEquals(2, u2.getUpdates().size());
```

Comment:
could also stream over the updates and filter deletes and then count and assert that count is 1
```suggestion
    assertEquals(2, u2.getUpdates().size());
    assertEquals(1, u2.getUpdates().stream().filter(cup->cup.isDeleted()).count());
```

Reply:
Added with e09da28032